processing/src/main/java/io/druid/segment/filter/OrFilter.java

```diff
@@ -103,23 +104,14 @@ public ValueMatcher makeMatcher(
       matchers.add(0, offsetMatcher);
     }
 
-    return new ValueMatcher()
-    {
-      @Override
-      public boolean matches()
-      {
-        for (ValueMatcher valueMatcher : matchers) {
-          if (valueMatcher.matches()) {
-            return true;
-          }
-        }
-        return false;
-      }
-    };
+    return makeMatcher(matchers.toArray(new ValueMatcher[0]));
   }
 
 
   private ValueMatcher makeMatcher(final ValueMatcher[] baseMatchers){
+    if (baseMatchers.length == 0) {
+      return BooleanValueMatcher.of(false);
```

Comment:
Identity ("zero") element of OR is logical `false`. If this check is not done, and empty array is used in the anonymous `ValueMatcher` below, it will always return `false`.